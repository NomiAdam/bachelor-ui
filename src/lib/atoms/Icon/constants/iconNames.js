import {
    IoAndroidAttach,
    IoIosCogOutline,
    IoIosAlbumsOutline,
    IoIosStar,
    IoIosStarOutline,
    IoAndroidAdd,
    IoIosUndoOutline,
    IoIosPhotosOutline,
    IoIosArrowBack,
    IoAperture,
    IoIosCloseOutline,
    IoAndroidSend,
    IoIosDownloadOutline,
    IoIosTrashOutline,
    IoEye,
    IoEyeDisabled,
    IoIosSearch,
    IoIosArrowDown,
} from 'react-icons/lib/io';
import { keys } from 'ramda';

const IconTypes = {
    cog: IoIosCogOutline,
    starFill: IoIosStar,
    star: IoIosStarOutline,
    documents: IoIosAlbumsOutline,
    attach: IoAndroidAttach,
    reply: IoIosUndoOutline,
    download: IoIosDownloadOutline,
    compose: IoIosPhotosOutline,
    aperture: IoAperture,
    arrowBack: IoIosArrowBack,
    arrowDown: IoIosArrowDown,
    add: IoAndroidAdd,
    close: IoIosCloseOutline,
    send: IoAndroidSend,
    trash: IoIosTrashOutline,
    eyeOff: IoEyeDisabled,
    eye: IoEye,
    search: IoIosSearch,
};

export const iconStringTypes = keys(IconTypes);

export default IconTypes;
