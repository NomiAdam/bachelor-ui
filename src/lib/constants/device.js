const size = {
    mobileS: '320px',
    mobileM: '375px',
    mobileL: '425px',
    tablet: '768px',
    laptop: '1024px',
    laptopL: '1440px',
    desktop: '2560px'
};

export const maxMobileS = `(max-width: ${size.mobileS})`;
export const maxMobileM = `(max-width: ${size.mobileM})`;

export const minTablet = `(min-width: ${size.tablet})`;
export const maxTablet = `(max-width: ${size.tablet})`;

export default {
    mobileS: `(min-width: ${size.mobileS})`,
    mobileM: `(min-width: ${size.mobileM})`,
    mobileL: `(min-width: ${size.mobileL})`,
    tablet: `(min-width: ${size.tablet})`,
    laptop: `(min-width: ${size.laptop})`,
    laptopL: `(min-width: ${size.laptopL})`,
    desktop: `(min-width: ${size.desktop})`,
    desktopL: `(min-width: ${size.desktop})`
};