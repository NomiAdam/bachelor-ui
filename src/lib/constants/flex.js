export default {
    FLEX: 'flex',
    DIRECTION_ROW: 'row',
    DIRECTION_COLUMN: 'column',
    POSITION_LEFT: 'flex-start',
    POSITION_RIGHT: 'flex-end',
    POSITION_CENTER: 'center',
    POSITION_SPACER: 'space-between',
    POSITION_AROUND: 'space-around',
};
