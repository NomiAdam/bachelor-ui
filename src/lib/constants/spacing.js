export const margin = {};

export const padding = {};
