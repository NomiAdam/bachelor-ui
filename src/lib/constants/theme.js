export const basicTheme = {
    blue: '#0247fe',
    red: '#ff230f',
    green: '#009933',
    purple: '#9a0794',
    orange: '#ff9933',
    grey: '#94989f',
};

export const lightTheme = {
    lightGrey: '#f5f5f5',
    white: '#ffffff',
    whiteBlue: '#c2eeff',
    lightBlue: '#0099ff',
    lightPurple: '#4424d6',
    lightGreen: '#00ad3e',
    lightOrange: '#ffbf00',
};

export const darkTheme = {
    darkBlue: '#110934',
    darkPurple: '#521751',
    darkGrey: '#1d1f1f',
    black: '#000000',
    darkOrange: '#ff4000'
};
