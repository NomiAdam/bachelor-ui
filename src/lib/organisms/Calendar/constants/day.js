export default [
    {
        items: [{
            id: 1,
            label: 'RoleOne',
            start: '6:00',
            end: '8:00',
        }],
    },
    {
        items: [],
    },
    {
        items: [{
            id: 2,
            label: 'RoleTwo',
            start: '7:00',
            end: '10:00',
        }],
    },
    {
        items: [],
    }
];

// TODO: Compute start and span a dates but calculated it differently
