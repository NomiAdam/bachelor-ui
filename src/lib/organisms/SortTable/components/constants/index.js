// eslint-disable-next-line import/prefer-default-export
export const basicRowCount = [
    {
        value: 10,
        label: 10,
    },
    {
        value: 25,
        label: 25,
    },
    {
        value: 50,
        label: 50,
    },
];
