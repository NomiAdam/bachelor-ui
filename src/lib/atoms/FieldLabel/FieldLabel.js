import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { basicTheme } from '../../constants/theme';

const StyledSpan = styled.span`
    font-size: .7rem;
    color: ${({error}) => error ? basicTheme.red : 'inherit'};
    display: inline-block;
    ${({hasText}) => hasText ? '' : 'height: 0px;'}
`;

const FieldLabel = ({ label, error }) => (
    <StyledSpan error={error} hasText={label.length > 0}>
        {label}
    </StyledSpan>
);

FieldLabel.propTypes = {
    /**
     * String representation of label text
     */
    label: PropTypes.string,
    /**
     * Whether we are displaying error label or not
     */
    error: PropTypes.bool,
};

FieldLabel.defaultProps = {
    label: '',
};

export { StyledSpan };
export default FieldLabel;
