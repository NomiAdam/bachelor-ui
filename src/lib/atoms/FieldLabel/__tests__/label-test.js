/* eslint-disable no-undef */
import React from 'react';
import FieldLabel, {StyledSpan} from '../FieldLabel';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import { basicTheme } from '../../../constants/theme';

describe('FieldLabel', () => {
    it('should be defined', () => {
        expect(FieldLabel).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<FieldLabel />)).toMatchSnapshot();
    });
    it('should be of type StyledSpan', () => {
        expect(shallow(<FieldLabel />).type()).toBe(StyledSpan);
    });
    it('should have red color when error', () => {
        const wrapper = shallow(<FieldLabel error />);
        expect(wrapper.find(StyledSpan)).toHaveStyleRule('color', basicTheme.red);
    });
    it('should have default color', () => {
        const wrapper = shallow(<FieldLabel />);
        expect(wrapper.find(StyledSpan)).toHaveStyleRule('color', 'inherit');
    });
});
