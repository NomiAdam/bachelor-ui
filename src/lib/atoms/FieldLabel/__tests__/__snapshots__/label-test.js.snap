// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldLabel should match the snapshot 1`] = `
<FieldLabel__StyledSpan
  hasText={false}
/>
`;
